components/theme-toggle.tsx

```diff
@@ -7,13 +7,24 @@ import { Button } from "@/components/ui/button"
 import { Icons } from "@/components/icons"
 
 export function ThemeToggle() {
-  const { setTheme, theme } = useTheme()
+  const { setTheme, theme } = useTheme();
 
+  let sysMode:String|null=null;
+
+  // Check if the system is in dark mode
+  if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
```

Comment:
This can't work, because `window` is not defined. Maybe try to use a[ isoMorphic hook](https://metaphore.vercel.app/stories/javascript/useisomorphiclayouteffect-react-hook), or do another integration.

Reply:
To resolve this issue I have used useState and useEffect hooks